File: solutions/204.java
class Solution {
    public int countPrimes(int n) {
        int c=0;
        boolean prine[]=new boolean[n];
        Arrays.fill(prine,true);
        
        for(int i=2;i<n;i++)
        {
            if(prine[i]==true)
            {
                for(int j=2;i*j<n;j++)
                {
                    prine[i*j]=false;
                }
            }
        }
        for(int i=2;i<n;i++)
        {
            if(prine[i]==true)
                c++;
        }
        return c;
    }
}

File: solutions/204.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  int countPrimes(int n) {
    if (n < 2) return 0;
    vector<bool> Prime(n + 1, true);

    Prime[0] = false;
    Prime[1] = false;

    Prime[n] = false;

    for (int p = 2; p * p <= n; p++) {
      if (Prime[p] == true) {
        for (int i = p * p; i <= n; i += p) {
          Prime[i] = false;
        }
      }
    }
    int ans = 0;
    for (auto element : Prime) {
      if (element) {
        ans++;
      }
    }
    return ans;
  }
};
